test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -257,31 +261,114 @@ private static void getFailureReason() {
+     *                  VERTICAL - the test instruction frame is positioned such
+     *                  that its bottom edge aligns with the screen's vertical
+     *                  center and the test window (if not null) is placed below
+     *                  the instruction frame.
+     *
+     *                  TOP_LEFT_CORNER - the test instruction frame is positioned
+     *                  such that its top left corner is at the top left corner of
+     *                  the screen and the test window (if not null) is placed to
+     *                  the right of the instruction frame.
      */
     public static void positionTestWindow(Window testWindow, Position position) {
         Dimension screenSize = Toolkit.getDefaultToolkit().getScreenSize();
+
+        // to get the screen insets inorder to position the frame by taking into
+        // account the location of taskbar/menubars on screen
+        GraphicsConfiguration gc = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration();
+        Insets screenInsets = Toolkit.getDefaultToolkit().getScreenInsets(gc);
+
         if (position.equals(Position.HORIZONTAL)) {
             int newX = ((screenSize.width / 2) - frame.getWidth());
-            frame.setLocation(newX, frame.getY());
-
-            testWindow.setLocation((frame.getLocation().x + frame.getWidth() + 5), frame.getY());
+            frame.setLocation((newX + screenInsets.left),
+                    (frame.getY() + screenInsets.top));
+            syncLocationToWindowManager();
+            if (testWindow != null) {
+                testWindow.setLocation((frame.getX() + frame.getWidth() + 5),
+                        frame.getY());
+            }
         } else if (position.equals(Position.VERTICAL)) {
             int newY = ((screenSize.height / 2) - frame.getHeight());
-            frame.setLocation(frame.getX(), newY);
+            frame.setLocation((frame.getX() + screenInsets.left),
+                    (newY + screenInsets.top));
+            syncLocationToWindowManager();
+            if (testWindow != null) {
+                testWindow.setLocation(frame.getX(),
+                        (frame.getY() + frame.getHeight() + 5));
+            }
+        } else if (position.equals(Position.TOP_LEFT_CORNER)) {
+            frame.setLocation(screenInsets.left, screenInsets.top);
+            syncLocationToWindowManager();
+            if (testWindow != null) {
+                testWindow.setLocation((frame.getX() + frame.getWidth() + 5),
+                        frame.getY());
+            }
+        }
+        // make instruction frame visible after updating
+        // frame & window positions
+        frame.setVisible(true);
+    }
+
+    /**
+     * To ensure the frame location is updated by the window manager
+     * if it adjusts the frame location after {@code setLocation}
```

Comment:
```suggestion
     * Ensures the frame location is updated by the window manager
     * if it adjusts the frame location after {@code setLocation}
```

For consistent documentation where method descriptions start with a verb.